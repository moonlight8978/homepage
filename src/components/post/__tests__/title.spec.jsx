import React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import Title, { LinkTitle } from '../title'

test('renders small title when big is omitted', () => {
  const { container } = render(<Title>Sample</Title>)
  expect(container.firstChild).toMatchSnapshot()
})

test('renders big title when big is passed', () => {
  const { container } = render(<Title big>Sample</Title>)
  expect(container.firstChild).toMatchSnapshot()
})

test('renders title with link', () => {
  const { container } = render(
    <MemoryRouter>
      <LinkTitle to="/">
        <Title>Sample</Title>
      </LinkTitle>
    </MemoryRouter>
  )
  expect(container.firstChild).toMatchSnapshot()
})
