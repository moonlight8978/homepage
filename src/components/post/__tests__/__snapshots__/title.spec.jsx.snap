// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders big title when big is passed 1`] = `
<h1
  class="title bigTitle"
>
  Sample
</h1>
`;

exports[`renders small title when big is omitted 1`] = `
<h2
  class="title"
>
  Sample
</h2>
`;

exports[`renders title with link 1`] = `
<a
  class="link"
  href="/"
>
  <h2
    class="title"
  >
    Sample
  </h2>
</a>
`;
